### PR TITLE
Pretender preventer

### PR DESCRIPTION
**Pull request by Cpt0r**
## TL;DR:

Fix skewer on pages which mess with XMLHttpRequest causing CORS issues - pretender, mirage, etc.

## What does it do?

1. Configures the user script to be loaded before the page document.
2. Captures and exposes the untampered native XMLHttpRequest object.
3. Exposes the injector function for manual activation - in case your ember page obliterates all divs that came before it (like mine seems to do).
4. Use the native XHR in POST/GET requests.

## Details

After spending more time than I would have liked trying to make pretender/mirage/ember to stop breaking by skewer requests I gave up and found a solution that will always work without messing with the server.

Specifically I was getting a missing `Access-Control-Allow-Origin` header error, in turn caused by a nil JSON block being sent back to emacs, throwing a `(json-end-of-file)` error, and thus not sending back a proper response.

...or something like that, I'm neither an elisp nor http expert.

## Anything left to do?

- You might like to add a simple bookmarklet which calls `skewer` with a single click.
- Not sure if `window.skewer` might conflict with `skewer.fn` in some way?

### CODE REVIEW

**Comment by skeeto** [comment]
Thanks! I was unaware of this issue. I've accepted all of your changes, 
but I made two additions:

* The "obliterates all divs that came before it" problem may have been 
because the div creation broke when the @run-at was changed. That's 
because document.body is now null when the script runs. I just put this 
in an event listener so it runs later, like before.

* I decided to "namespace" the new window property by prefixing it with 
"skewer". And yes, the name "skewer" was conflicting with the name from 
skewer.js, so I renamed it to skewerInject.

> You might like to add a simple bookmarklet which calls `skewer` with a 
> single click.

There's a bookmarklet in the README that does this without even needing 
the userscript.

**Comment by Cpt0r** [comment]
Ah yea that makes sense about the div.

Cheers!
